v2/tools/generator/internal/astmodel/property_definition.go

```diff
@@ -156,6 +158,17 @@ func (property *PropertyDefinition) SetFlatten(flatten bool) *PropertyDefinition
 	return &result
 }
 
+// SetSecret sets if the property is a secret
+func (property *PropertyDefinition) SetSecret(secret bool) *PropertyDefinition {
```

Comment:
"Set" implies mutability - this should probably be a `With` method, in line with the others.

Reply:
Yeah I was following the naming for `SetFlatten`, but agree it doesn't match most of our other naming

Reply:
Changed this